src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -231,31 +265,26 @@ void SensorFusion::updateWorld(const SSLProto::SSL_DetectionFrame &ssl_detection
         ball_in_dribbler_timeout                = 0;
     }
 
-    if (friendly_robot_id_with_ball_in_dribbler.has_value())
+    if (shouldTrustRobotStatus(ssl_detection_frame, ball_detections))
     {
         std::optional<Robot> robot_with_ball_in_dribbler =
```

Comment:
in the `else` case of this `if` statement:
reset `friendly_robot_id_with_ball_in_dribbler` to nullopt and set `ball_in_dribbler_timeout` back to 0

Reply:
done